CLAUDE.md

```diff
@@ -0,0 +1,331 @@
+source .venv/bin/activate
+
+# Run Chainlit development server
+uv run chainlit run app.py -w
+```
+
+### Frontend Commands
+```bash
+# Development server with hot reload
+npm run dev          # http://localhost:5173
+
+# Production build
+npm run build        # Output to src/frontend/dist/
+
+# Preview production build
+npm run preview
+
+# Run linter
+npm run lint
+```
+
+### Running Backend Tests
+```bash
+# Run a single test file
+uv run pytest tests/test_agent.py
+
+# Run tests matching a pattern
+uv run pytest -k "test_name_pattern"
+
+# Run with verbose output
+uv run pytest -v
+
+# Run specific test markers
+uv run pytest -m agent  # Same as --agent flag
+```
+
+## Architecture
+
+### Frontend Architecture
+
+The React frontend provides a modern web interface with:
+
+- **Three-column layout**: Logo/avatar sidebar, collapsible prompt builder, main content area
+- **Component-based architecture**: Modular React components with TypeScript
+- **CSS Modules**: Scoped styling with design tokens
+- **React 19 features**: React Compiler for automatic optimization
+- **Vite bundler**: Fast development and optimized production builds
+
+See [src/frontend/CLAUDE.md](./src/frontend/CLAUDE.md) for detailed frontend documentation.
+
+### Backend Architecture (LangGraph Agent)
+
+The backend uses LangGraph's `create_react_agent` with a custom state schema and two primary tools:
+
+1. **select_dataset** - Searches ChromaDB vector store (Mistral embeddings) to find matching climate datasets from `data/datasets.json`
+2. **create_chart** - Generates SQL queries (via Codestral) to extract data from S3 parquet files, then creates Plotly visualizations
+
+**Agent Flow:**
+```
+User Query → Agent (Mistral chat model) → Tool Selection
+                                        ↓
+                            select_dataset (ChromaDB similarity search)
+                                        ↓
+                            create_chart (Codestral generates SQL → DuckDB execution → Codestral generates Plotly code)
+                                        ↓
+                            Visualization + Response
+```
+
+### State Management
+
+Backend state via custom `AgentState` (extends `AgentStatePydantic`):
+- `dataset`: Selected dataset metadata from ChromaDB
+- `chart_query`: Generated SQL query for data extraction
+- `python_code`: Generated Plotly visualization code
+- `chart_data`: Processed data for visualization
+- `chart`: Final Plotly chart JSON
+
+State is persisted via `InMemorySaver` checkpointer with thread-based conversation history.
+
+### Code Structure
+
+```
+# Backend Structure
+src/atlas_assistant/
+├── agent.py           # LangGraph agent creation and system prompt
+├── state.py          # AgentState schema definition
+├── settings.py       # Pydantic settings with Mistral API config
+└── tools/
+    ├── select_dataset.py    # ChromaDB-based dataset retrieval
+    └── create_chart.py      # SQL generation + Plotly visualization
+
```

Comment:
I know I'm a dinosaur 🦖, but I get a little weirded out by this section. I don't want to have to update this every time I move backend files around?